SPECS/ksh/ksh.spec

```diff
@@ -80,12 +71,14 @@ touch %{buildroot}%{_bindir}/rksh
 touch %{buildroot}%{_mandir}/man1/rksh.1.gz
 
 %check
-# We run more tests as non-root user
-chmod g+w . -R
 useradd test -G root -m
 
-# Disabling tests as they tend to freez and the package is low-pri at the moment.
-false && sudo -u test ./bin/shtests --compile
+# disable pty tests as they tend to freeze
+shopt -s globstar; shopt -s nullglob; for test in bin/tests/**/*.sh; do
+  if [[ $test != *"pty.sh" ]]; then
+    $test || exit 1;
```

Comment:
We should never `exit` out of the `%check` section. This breaks the whole package build and not just indicate a failure of the tests. The way test failures are identified is the return value of the last command. If there's more than one test to run, you could for instance have a helper `tests_ok` variable, which you set to `false` whenever we detect a failing test and as the last line in the `%check` section to `$tests_ok`. Example in `edk2.spec`.

Having said that, please see my comment below about using upstream's way of running tests, which may make this comment irrelevant.